src/main/java/uk/gov/companieshouse/service/dissolution/certificate/DissolutionCertificateGenerator.java

```diff
@@ -32,7 +35,8 @@ public DissolutionCertificateGenerator(
     public DissolutionCertificate generateDissolutionCertificate(Dissolution dissolution) {
         final DissolutionCertificateData data = mapper.mapToCertificateData(dissolution);
 
-        final String location = locationGenerator.generateCertificateLocation();
+        final String certificateFileName = CERTIFICATE_FILE_NAME_PREFIX + dissolution.getData().getApplication().getReference();
```

Comment:
don't ya think this should go inside the location generator? Just pass the dissolution and let it extract the necessary data it needs i.e. reference.

Reply:
Originally I was thinking to keep this generic and just take a single value (and let other classes do the work). But makes sense

Reply:
Would make sense if it was just `LocationGenerator`, but it's `DissolutionLocationGenerator` so I think it's fine.